boards/common/nucleo/include/periph_adc_144_ac.h

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (C) 2017 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     boards_common_nucleo
+ * @brief       XX
+ * @{
+ *
+ * @file
+ * @brief       XXX
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ */
+
+#ifndef PERIPH_ADC_144_AC_H
```

Comment:
Why naming it `periph_..144_ac" instead of putting it in the `nucleo144/include` directory ?
Maybe I am missing something.

Reply:
This still needs to be sorted out, I have the feeling that there won't be a nucleo144/include once this effort ist done...

Reply:
It's because of the corresponding port of the configured ADCs. For those type of boards, you have 2, one on port A and one on port C.

Reply:
some other has 6 => aaaccc (or something like this)

Reply:
@aabadie was refering to the `144` in particular, but it was not clear.

Reply:
as stated above: let's not dive into these kind of specific questions (yet), as the conversion is far from complete and not everything is figured out yet...

Reply:
t

Reply:
While the Nucleo-X boards do undeniably all support ADC, it is just wrong to support the ADC peripheral at BOARD level in any way as the ADC in Nucleos are always feature of the MCU.

Some Nucleo boards do have peripherals that are external to the MCU and supporting them at the BOARD level files is proper and good. The clock resonator configuration is a great example of something that is a Nucleo BOARD level feature. Other very good examples of BOARD features in Nucleo are the buttons and LEDs. Some Nucleos even have network chips and acceleration sensors and the like soldered on board.

Also more about ADC: since the Nucleo family is specifically designed to map out _all_ MCU pins, all of the Nucleo boards support all the peripherals that the MCU happens to have, which means there are multiple multichannel ADC devices, hordes of multiple types if timers, SPI, I2C, CEC, some have LCD drivers etc etc.

Most of these peripherals are features of MCU and thus discussing WHICH BOARD level file should support them is moot since the answer is: none. They should be supported in MCU files.

Reply:
> Most of these peripherals are features of MCU and thus discussing WHICH BOARD level file should support them is moot since the answer is: none. They should be supported in MCU files.

I'd say this is only partly correct, as there are some implications to consider:
- all the peripherals that can be mapped to alternative pin mappings: the board configuration **must** specify which mapping to use (how should the generic CPU code know this?) (e.g. SPI, UART, ...)
- all peripherals that use resources (e.g. RAM for state): the board configuration **must** specify which ones and how many of them to use (e.g. timers, UART, ...)

Now (for simplicity let's stay in the domain of STM CPUs) the ADC falls into the second category, as the driver at least allocates memory for 1 mutex per ADC device -> so IMHO, there has to be some kind of configuration in the board!